themes-default/slim/views/vue-components/name-pattern.mako

```diff
@@ -0,0 +1,491 @@
+            */
+            enabled: {
+                type: Boolean,
+                default: true
+            }
+        },
+        data() {
+            return {
+                presets: [],
+                availableMultiEpStyles: [],
+                pattern: '',
+                customName: '',
+                showLegend: false,
+                namingExample: '',
+                namingExampleMulti: '',
+                isEnabled: false,
+                isMulti: false,
+                selectedMultiEpStyle: 1,
+                animeType: 0
+            }
+        },
+        methods: {
+            getDateFormat(format) {
+                return dateFns.format(new Date(), format);
+            },
+            async testNaming(pattern, selectedMultiEpStyle, animeType) {
+                console.debug('Test pattern ' + pattern + ' for ' + (selectedMultiEpStyle) ? 'multi' : 'single' + ' ep');
+                const params = {
+                    pattern: pattern,
+                    anime_type: animeType
+                }
+
+                if (selectedMultiEpStyle) {
+                    params.multi = selectedMultiEpStyle;
+                }
+
+                try {
+                    const result = await apiRoute.get('config/postProcessing/testNaming', {params: params});
+                    return result.data;
+                } catch (e) {
+                    console.warning(e);
+                }
+            },
+            async updatePatternSamples() {
+                // If it's a custom pattern, we need to get the custom pattern from this.customName
+                const pattern = (this.isCustom) ? this.customName : this.pattern; 
+                
+                // Update single
+                this.namingExample = await this.testNaming(pattern, false, this.animeType) + '.ext';
+                console.debug('Result of naming pattern check: ' + this.namingExample);
+
+                // Test naming
+                this.checkNaming(pattern, false, this.animeType);
+                
+                // Update multi if needed
+                if (this.isMulti) {
+                    this.namingExampleMulti = await this.testNaming(pattern, this.selectedMultiEpStyle, this.animeType) + '.ext';
+                    this.checkNaming(pattern, this.selectedMultiEpStyle, this.animeType);
+                }
+            },
+            update(newValue) {
+                this.$emit('change', {
+                    pattern: (this.isCustom) ? this.customName : this.pattern,
+                    type: this.type,
+                    multiEpStyle: this.selectedMultiEpStyle,
+                    custom: this.isCustom,
+                    enabled: this.isEnabled,
+                    animeNamingType: this.animeType
+                });
+            },
+            async checkNaming(pattern, selectedMultiEpStyle, animeType) {
```

Comment:
Again with the async, there are places above that don't use await so they'll likely remain as a unresolved promise until you do something with them.